### PR TITLE
implement eXistdb REST-api client

### PR DESCRIPTION
**Pull request by line-o**
node-exist exports a new async function `getRestClient` accepting the
same connection options as connect did.
The returned client has three methods `get`, `post`, `put` and `del` to
interact with eXist-db using the [REST-API](https://exist-db.org/exist/apps/doc/devguide_rest)

```
const { getRestClient } = require('../../index')
const rc = await getRestClient(connectionOptions)
await rc.put('<root />', '/db/test.xml')
await rc.get('/db/test.xml')
await rc.post('/db', query, options)
await rc.del('/db/test.xml')
```

see [example/rest.mjs](https://github.com/eXist-db/node-exist/pull/243/files#diff-3a44cd83eda759197802110e8a6dcf7c828edf223c78dc63421217a154384511) for a runnable version of above code

### CODE REVIEW

File: spec/examples/exist-upload [hunk 1]
```diff
@@ -292,7 +304,7 @@ Usage:
       console.log("target exists:", existsAndCanOpenCollection)
     }
 
-    return uploadFileOrFolder(db, source, target, parsed)
+    return uploadFileOrFolder(db, rc, source, target, parsed)
   }
   catch (error) {
     if (error.name !== 'YError') {
```
Comment (duncdrum) [inline spec/examples/exist-upload:310]: is `yerror` provided by `yargs` ? do we need an import here, typo?

**Review by duncdrum — COMMENTED** [review]
overall the api design looks good to me. Very nice feature. `Get`ing collection listing in JSON might be useful, but its something i would add once we have a usecase that isn't served by `xst`. `Post`ing xquerys on the other hand is something that I personally would have use for, it would be nice to add that. 

The Roadmap on the Readme should be updated with this PR, and it needs some documentation / sample code. 

I would add another test on `specs/rest.js` to ensure that exist reports the correct mimetypes for uploaded txt and xml files. This is an area of frequent problems.

File: spec/examples/exist-upload (hunk 1 above)
Comment (line-o) [reply spec/examples/exist-upload:310]: Correct `YError` is the error type from yargs. As it is a string I am not sure what to import here.

**Comment by line-o** [comment]
@duncdrum added POST, more tests, documentation and the feature server responses that are wrapped in `<exist:result>` will now extend the returned response with some useful metadata extracted (start, hits, count, ....)

**Review by duncdrum — APPROVED** [review]

**Comment by duncdrum** [comment]
@line-o feel free to merge after rebasing, the `rebase and merge` button is greyed out for me, so  i guess you need to resolve some conflict first

**Pull request merged**